analysis_driver/driver.py

```diff
@@ -366,6 +383,145 @@ def qc_pipeline(dataset, species):
+
+    basename = os.path.join(gatk_run_dir, user_sample_id)
+    sorted_bam = os.path.join(cfg['jobs_dir'], sample_id, sample_id + '.bam')
+    recal_bam = basename + '_recal.bam'
+    output_grp = basename + '.grp'
+    output_intervals = basename + '.intervals'
+    indel_realigned_bam = basename + '_indel_realigned.bam'
+    sample_gvcf = basename + '.g.vcf'
+    sample_gvcfgz = sample_gvcf + '.gz'
+
+    base_recal = executor.execute(
+        gatk_cmd('BaseRecalibrator', sorted_bam, output_grp, ext='--knownSites ' + dbsnp),
+        job_name='gatk_base_recal',
+        working_dir=gatk_run_dir,
+        cpus=16,
+        mem=16
+    ).join()
+
+    print_reads = executor.execute(
+        gatk_cmd('PrintReads', sorted_bam, recal_bam, ext=' -BQSR ' + output_grp),
+        job_name='gatk_print_reads',
+        working_dir=gatk_run_dir,
+        cpus=16,
+        mem=16
+    ).join()
+
+    realign_target_cmd = gatk_cmd('RealignerTargetCreator', recal_bam, output_intervals, nct=1)
+    if known_indels:
+        realign_target_cmd += ' --known ' + known_indels
+    realign_target = executor.execute(
+        realign_target_cmd,
+        job_name='gatk_realign_target',
+        working_dir=gatk_run_dir,
+        mem=16
+    ).join()
+
+    realign_cmd = gatk_cmd(
+        'IndelRealigner',
+        recal_bam,
+        indel_realigned_bam,
+        nct=1,
+        ext='-targetIntervals ' + output_intervals
+    )
+    if known_indels:
+        realign_cmd += ' --knownAlleles ' + known_indels
+    realign = executor.execute(
+        realign_cmd,
+        job_name='gatk_indel_realign',
+        working_dir=gatk_run_dir,
+        mem=16
+    ).join()
+
+    haplotype_cmd = gatk_cmd(
+        'HaplotypeCaller',
+        indel_realigned_bam,
+        sample_gvcf,
+        ext=('--pair_hmm_implementation VECTOR_LOGLESS_CACHING -ploidy 2 --emitRefConfidence GVCF '
+             '--variant_index_type LINEAR --variant_index_parameter 128000 --dbsnp ' + dbsnp)
+    )
+    for annot in ('BaseQualityRankSumTest', 'FisherStrand', 'GCContent', 'HaplotypeScore', 'HomopolymerRun',
+                  'MappingQualityRankSumTest', 'MappingQualityZero', 'QualByDepth', 'ReadPosRankSumTest',
+                  'RMSMappingQuality', 'DepthPerAlleleBySample', 'Coverage', 'ClippingRankSumTest',
+                  'DepthPerSampleHC'):
+        haplotype_cmd += ' --annotation ' + annot
+    haplotype = executor.execute(
+        haplotype_cmd,
+        job_name='gatk_haplotype_call',
+        working_dir=gatk_run_dir,
+        cpus=16,
+        mem=16
+    ).join()
+
+    bgzip = executor.execute(
+        '%s %s' % (cfg['tools']['bgzip'], sample_gvcf),
+        job_name='bgzip',
+        working_dir=gatk_run_dir,
+        cpus=16,
+        mem=16
+    ).join()
+    tabix = executor.execute(
+        '%s -p vcf %s' % (cfg['tools']['tabix'], sample_gvcfgz),
+        job_name='tabix',
+        working_dir=gatk_run_dir,
+        cpus=16,
+        mem=16
+    ).join()
+
```

Comment:
bgzip and tabix are single threaded so no need to give 16 cpus 
mem should alsp be reduce to 4G